src/Microsoft.SqlTools.ServiceLayer/Scripting/ScriptAsScriptingOperation.cs

```diff
@@ -106,18 +137,323 @@ public override void Execute()
-                        ErrorMessage = e.Message,
+                        ErrorMessage = $"{SR.ScriptingGeneralError} {e.Message}",
                         ErrorDetails = e.ToString(),
                     });
                 }
             }
             finally
+            {
+                if (disconnectAtDispose && ServerConnection != null && ServerConnection.IsOpen)
+                {
+                    ServerConnection.Disconnect();
+                }
+            }
+        }
+
+        private string GenerateScriptSelect(Server server, UrnCollection urns)
+        {
+            string script = string.Empty;
+            ScriptingObject scriptingObject = this.Parameters.ScriptingObjects[0];
+            Urn objectUrn = urns[0];
+            string typeName = objectUrn.GetNameForType(scriptingObject.Type);
+
+            // select from service broker
+            if (string.Compare(typeName, "ServiceBroker", StringComparison.CurrentCultureIgnoreCase) == 0)
+            {
+                script = Scripter.SelectAllValuesFromTransmissionQueue(objectUrn);
+            }
+
+            // select from queues
+            else if (string.Compare(typeName, "Queues", StringComparison.CurrentCultureIgnoreCase) == 0 ||
+                     string.Compare(typeName, "SystemQueues", StringComparison.CurrentCultureIgnoreCase) == 0)
+            {
+                script = Scripter.SelectAllValues(objectUrn);
+            }
+
+            // select from table or view
+            else
+            {
+                Database db = server.Databases[databaseName];
+                bool isDw = db.IsSqlDw;
+                script = new Scripter().SelectFromTableOrView(server, objectUrn, isDw);
+            }
+
+            return script;
+        }
+
+        private string GenerareScriptAsExecute(Server server, UrnCollection urns, ScriptingOptions options)
+        {
+            string script = string.Empty;
+            ScriptingObject scriptingObject = this.Parameters.ScriptingObjects[0];
+            Urn urn = urns[0];
+
+            // get the object
+            StoredProcedure sp = server.GetSmoObject(urn) as StoredProcedure;
+
+            Database parentObject = server.GetSmoObject(urn.Parent) as Database;
+
+            StringBuilder executeStatement = new StringBuilder();
+
+            // list of DECLARE <variable> <type>
+            StringBuilder declares = new StringBuilder();
+            // Parameters to be passed
+            StringBuilder parameterList = new StringBuilder();
+            if (sp == null || parentObject == null)
+            {
+                throw new InvalidOperationException(SR.ScriptingExecuteNotSupportedError);
+            }
+            WriteUseDatabase(parentObject, executeStatement, options);
+
+            // character string to put in front of each parameter. First one is just carriage return
+            // the rest will have a "," in front as well.
+            string paramListPreChar = "\r\n   ";
```

Comment:
>\r\n [](start = 39, length = 4)

you may use Environment.NewLine instead of \r\n so that the newlines are generated properly for the platform.